toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java

```diff
@@ -297,14 +307,21 @@ private ConstructorInjectionTarget createConstructorInjectionTarget(TypeElement
       }
     }
 
+    final String cannotCreateAFactoryMessage = " Toothpick can't create a factory for it." //
+        + " If this class is itself a DI entry point (i.e. you call TP.inject(this) at some point), " //
+        + " then you can remove this warning by adding @SuppressWarnings(\"Injectable\") to the class." //
+        + " A typical example is a class using injection to assign its fields, that calls TP.inject(this)," //
+        + " but it needs a parameter for its constructor and this parameter is not injectable.";
+
     //search for default constructor
     for (ExecutableElement constructorElement : constructorElements) {
       if (constructorElement.getParameters().isEmpty()) {
         if (constructorElement.getModifiers().contains(Modifier.PRIVATE)) {
           if (!isInjectableWarningSuppressed(typeElement)) {
-            warning(constructorElement, //
-                "The class %s has a private default constructor, Toothpick can't create a factory for it.",
-                typeElement.getQualifiedName().toString());
+              String message = format("The class %s has a private default constructor. "
+                      + cannotCreateAFactoryMessage, //
+                  typeElement.getQualifiedName().toString());
+            crashOrWarnWhenNoFactoryCanBeCreated(constructorElement, message);
           }
           return null;
         }
```

Comment:
we are repeating a bunch of code and looks complicated. Could we clean it a little bit on this PR?:
```java
for (ExecutableElement constructorElement : constructorElements) {
    if (constructorElement.getParameters().isEmpty()) {
        if(constructorElement.getModifiers().contains(Modifier.PRIVATE)) {
            break;
        }
        return new ConstructorInjectionTarget(typeElement, scopeName, hasScopeInstancesAnnotation, superClassWithInjectedMembers);
    }
}

if (!isInjectableWarningSuppressed(typeElement)) {
...